### DISCUSSION TITLE
Rhino compatibility?

### DISCUSSION THREAD

**Discussion by phasinghawk** [openpgpjs/openpgpjs#1438] [Q&A]
Hi all,

I am trying to get OpenPGP to work with ServiceNow, which for some ungodly reason still runs on Rhino which is ES5.

Which older version of this module would be best for compatibility - has anyone else run with this use case?

Thanks

**Comment by twiss** [comment]
Hello :wave: OpenPGP.js v4 had a compatibility build that runs on ES5, but it seems unlikely to me that it (or any version of OpenPGP.js) will run on Rhino out of the box, as Rhino doesn't have support for Web Crypto. The most critical requirement is being able to get cryptographically secure random data. If you are somehow able to get that, and polyfill the Web Crypto API, you might be able to get it to work.
